src/components/application_manager/src/resumption/resumption_data_processor.cc

```diff
@@ -317,17 +331,14 @@ void ResumptionDataProcessor::ProcessResponseFromHMI(
     application_manager_.state_controller().DropPostponedWindows(app_id);
   }
 
-  resumption_status_lock_.AcquireForWriting();
+  sync_primitives::AutoWriteLock status_lock(resumption_status_lock_);
```

Comment:
Please reduce scope of the lock. 
Nested  multiple locks is strait way to dead locks

Reply:
This operations are enclosed in separate function, please look at 90d5b8f7a4d8ee35f2581eff4cb63a2126a04f08